### PR TITLE
phpPackages: phpmd

### PR DESCRIPTION
**Pull request by ccellado**
###### Motivation for this change

Add phpmd package.

###### Things done

Package added.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -202,7 +202,34 @@ in
...
+        mkdir -p $out/bin
+        install -D $src $out/libexec/phpmd/phpmd.phar
+        makeWrapper ${php}/bin/php $out/bin/phpmd \
+          --add-flags "$out/libexec/phpmd/phpmd.phar"
+      '';
+
+      meta = with pkgs.lib; {
+        description = "PHP coding quality analizer";
+        license = licenses.bsd3;
+        homepage = "https://github.com/phpmd/";
```
Comment (etu) [inline pkgs/top-level/php-packages.nix:228]: I would say that `https://phpmd.org/` would be more correct as website

Comment (ccellado) [reply pkgs/top-level/php-packages.nix]: Got it, changed.

**Comment by ajs124** [comment]
Please rebase instead of reverting and re-reverting.

**Comment by etu** [comment]
This should be squashed (using rebase as @ajs124 said) to a single commit with a commitmessage along the lines: `php.packages.phpmd: init at X.Y.Z`

**Comment by ccellado** [comment]
> This should be squashed (using rebase as @ajs124 said) to a single commit with a commitmessage along the lines: `php.packages.phpmd: init at 2.8.2`

Done.

File: pkgs/top-level/php-packages.nix [hunk 2]
```diff
@@ -202,7 +202,34 @@ in
...
+        install -D $src $out/libexec/phpmd/phpmd.phar
+        makeWrapper ${php}/bin/php $out/bin/phpmd \
+          --add-flags "$out/libexec/phpmd/phpmd.phar"
+      '';
+
+      meta = with pkgs.lib; {
+        description = "PHP code quality analyzer";
+        license = licenses.bsd3;
+        homepage = "https://phpmd.org/";
+        maintainers = teams.php.members;
```
Comment (etu) [inline pkgs/top-level/php-packages.nix:229]: This phar file seems to throw errors on php73 and php72, so we should probably mark it as broken for those versions.

The following should do it nicely:
```suggestion
        maintainers = teams.php.members;
        broken = !isPhp74;
```

**Pull request merged**